assets/js/theme/global/compare-products.js

```diff
@@ -32,6 +32,10 @@ function decrementCounter(itemID) {
         const url = new URL(pathname)
         window.history.replaceState(null, '', url)
         q$(`[${selectors.dataCompareCard}='${itemID}']`).remove()
+
+        if (compareProductsIDs.length === 0) {
+            window.location.reload()
```

Comment:
Hmm, why do you need to reload the page? 🤔
It will cause the loading of all resources again, which is not great from the performance POV.

Reply:
How about removing `.c-product-compare` update heading's content or toggle some `visibility class` on the container, to display either "no results" heading or "product-compare" container? 

You wouldn't need to reload the page in such case 🤔